From f57c9f069dcf49df63469ab9255fa2e82007b162 Mon Sep 17 00:00:00 2001
From: Hari Chandrakanthan <quic_haric@quicinc.com>
Date: Sun, 6 Feb 2022 15:02:19 +0530
Subject: [PATCH] sigma dut : make h2e as default sae password element method

WPA3 mandates H2E to be used as password element method for sae.

Signed-off-by: Hari Chandrakanthan <quic_haric@quicinc.com>
---
 ap.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/ap.c b/ap.c
index cca7417..7e4b09d 100644
--- a/ap.c
+++ b/ap.c
@@ -2451,6 +2451,8 @@ static enum sigma_cmd_result cmd_ap_set_security(struct sigma_dut *dut,
 			dut->ap_key_mgmt = AP_WPA2_SAE;
 			dut->ap_cipher = AP_CCMP;
 			dut->ap_pmf = AP_PMF_REQUIRED;
+			if (dut->program == PROGRAM_HE)
+				dut->sae_pwe = SAE_PWE_H2E;
 		} else if (strcasecmp(val, "WPA2-PSK-SAE") == 0) {
 			dut->ap_key_mgmt = AP_WPA2_PSK_SAE;
 			dut->ap_cipher = AP_CCMP;
-- 
2.7.4

